rpcclient/infrastructure.go

```diff
@@ -1733,28 +1740,38 @@ func (c *Client) Send() error {
 		return nil
 	}
 
-	// clear batchlist in case of an error
-	defer func() {
+	batchResp, err := c.sendAsync().Receive()
+	if err != nil {
+		// Clear batchlist in case of an error.
+		//
+		// TODO(yy): need to double check to make sure there's no
+		// concurrent access to this batch list, otherwise we may miss
+		// some batched requests.
```

Comment:
yeah - and I guess we also miss all the ones that were in the list that didnt necessarily cause the failure...

Reply:
yeah, need more love in this lib, more tests, etc